Description: Display more helpful output when failing to load a device
 If the device is known to be one of the devices in the groff binary package
 rather than groff-base, refer the user to that.
Author: Colin Watson <cjwatson@debian.org>
Forwarded: not-needed
Last-Update: 2013-01-07

Index: b/src/roff/groff/groff.cpp
===================================================================
--- a/src/roff/groff/groff.cpp
+++ b/src/roff/groff/groff.cpp
@@ -333,8 +333,16 @@
     commands[PIC_INDEX].append_arg("-U");
   }
   font::set_unknown_desc_command_handler(handle_unknown_desc_command);
-  if (!font::load_desc())
-    fatal("invalid device `%1'", device);
+  if (!font::load_desc()) {
+    if (strcmp(device, "X100") == 0 || strcmp(device, "X100-12") == 0 ||
+	strcmp(device, "X75") == 0 || strcmp(device, "X75-12") == 0 ||
+	strcmp(device, "dvi") == 0 || strcmp(device, "html") == 0 ||
+	strcmp(device, "lbp") == 0 || strcmp(device, "lj4") == 0)
+      fatal("invalid device `%1' (try installing the `groff' package?)",
+	    device);
+    else
+      fatal("invalid device `%1'", device);
+  }
   if (!postdriver)
     fatal("no `postpro' command in DESC file for device `%1'", device);
   if (predriver && !zflag) {
